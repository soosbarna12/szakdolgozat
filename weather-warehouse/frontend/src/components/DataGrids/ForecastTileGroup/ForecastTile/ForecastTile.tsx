import * as React from 'react';
import { StyledItem } from '../../../../stlyes/content.style';

export default function ForecastTile({ dayName, temperature }: { dayName: string; temperature: number }) {
  return (
    <StyledItem data-testid="forecastTile">
      <p>{dayName}</p>
      <p>{temperature}Â°C</p>
    </StyledItem>
  );
};