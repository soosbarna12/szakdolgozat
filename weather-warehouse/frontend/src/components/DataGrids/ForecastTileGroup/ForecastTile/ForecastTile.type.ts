import { ForecastData } from "../../../../types/forecastData.type";

export interface ForecastTileProps {
  data: ForecastData;
  isLoading?: boolean;
}