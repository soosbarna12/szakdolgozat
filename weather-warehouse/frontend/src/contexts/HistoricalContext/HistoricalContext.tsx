import React, { createContext, useEffect, useState } from "react";
import { LOCAL_STORAGE_TEMPERATURE_SCALE } from "../../consts/temperatureScale.const";
import { TemperatureScale } from "../../types/temperatureScale.type";
import { HistoricalContextType, HistoricalLocationData, HistoricalPageData, HistoricalProviderProps } from "./HistoricalContext.type";


export const HistoricalContext = createContext<HistoricalContextType>({
  location: { name: "", lat: 0, lon: 0 },
  setLocation: () => { },
  historicalPageData: [] as any,
  setHistoricalPageData: () => { },
  temperatureScale: TemperatureScale.Celsius,
  setTemperatureScale: () => { },
});

export const HistoricalLocationProvider = ({ children }: HistoricalProviderProps) => {

  const [location, setLocation] = useState<HistoricalLocationData>(() => {
    const storedLocation = localStorage.getItem("historicalLocation");
    const defaultLocation = { name: "", lat: 0, lon: 0 };
    try {
      return storedLocation ? JSON.parse(storedLocation) : defaultLocation;
    } catch (error) {
      console.error("Failed to parse location from localStorage:", error);
      return defaultLocation;
    }
  });

  const [historicalPageData, setHistoricalPageData] = useState<HistoricalPageData[]>(() => {
    const storedData = localStorage.getItem("historicalPageData");
    try {
      return storedData ? JSON.parse(storedData) : [];
    } catch (error) {
      console.error("Failed to parse historical data from localStorage:", error);
      return [];
    }
  });

  const [temperatureScale, setTemperatureScale] = useState<TemperatureScale | string>(() => {
    const storedTemperatureScale = localStorage.getItem(LOCAL_STORAGE_TEMPERATURE_SCALE);
    try {
      return storedTemperatureScale ? JSON.parse(storedTemperatureScale) : TemperatureScale.Celsius;
    } catch (error) {
      console.error("Failed to parse temperature scale from localStorage:", error);
      return TemperatureScale.Celsius;
    }
  });


  useEffect(() => {
    if (historicalPageData) {
      localStorage.setItem("historicalPageData", JSON.stringify(historicalPageData));
    }
  }, [historicalPageData]);

  useEffect(() => {
    localStorage.setItem("historicalLocation", JSON.stringify(location));
  }, [location]);

  useEffect(() => {
    localStorage.setItem(LOCAL_STORAGE_TEMPERATURE_SCALE, JSON.stringify(temperatureScale));
  }, [temperatureScale]);


  return (
    <HistoricalContext.Provider value={{ location, setLocation, historicalPageData, setHistoricalPageData, temperatureScale, setTemperatureScale }}>
      {children}
    </HistoricalContext.Provider>
  );
};
