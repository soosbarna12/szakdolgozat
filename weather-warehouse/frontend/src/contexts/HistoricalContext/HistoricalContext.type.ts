import { Dispatch, ReactNode, SetStateAction } from "react";
import { HistoricalDataTable } from "../../types/historicalDataTable.type";
import { TemperatureScale } from "../../types/temperatureScale.type";

export interface HistoricalLocationData {
  name: string;
  country?: string;
  lat: number;
  lon: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-object-type
export interface HistoricalPageData extends HistoricalDataTable {}

export interface HistoricalContextType {
  location: HistoricalLocationData;
  historicalPageData: any [];
  temperatureScale: TemperatureScale | string;
  setLocation: Dispatch<SetStateAction<HistoricalLocationData>>;
  setHistoricalPageData: Dispatch<SetStateAction<HistoricalPageData[]>>;
  setTemperatureScale: Dispatch<SetStateAction<TemperatureScale | string>>;
}

export interface HistoricalProviderProps {
  children: ReactNode;
}
