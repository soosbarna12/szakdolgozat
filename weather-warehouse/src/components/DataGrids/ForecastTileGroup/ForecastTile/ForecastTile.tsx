import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid2';
import * as React from 'react';
import { StyledForecastTypography, StyledItem } from '../../../../common.style';

const Days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];

export default function ForecastPage() {
  return (
    <Box sx={{ flexGrow: 1, margin: "10px 200px 0 200px" }}>
      <Grid container columns={7} spacing={2} justifyContent="center" alignItems="center">
        {Days.map((day) => (


          <Grid key={day} size={{ lg: 1 }}>
            <StyledItem elevation={4}>
              <Box
                id="category-c"
                sx={{ fontSize: '12px', textTransform: 'uppercase' }}
              >
                <StyledForecastTypography variant='h6' sx={{ textAlign: "center" }}>
                  {day}
                </StyledForecastTypography>
              </Box>
              <Box component="ul" aria-labelledby="category-c" sx={{ pl: 2 }}>

              </Box>
            </StyledItem>
          </Grid>


        ))}
      </Grid>
    </Box>
  );
}