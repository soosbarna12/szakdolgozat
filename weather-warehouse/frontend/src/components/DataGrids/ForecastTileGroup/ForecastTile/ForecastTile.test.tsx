import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { ForecastTile } from './ForecastTile';
import { TemperatureScale } from '../../../../types/temperatureScale.type';
import { HistoricalContext } from '../../../../contexts/HistoricalContext/HistoricalContext';

// Mock react-leaflet components
jest.mock('react-leaflet', () => ({
  MapContainer: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dataMapContainer">{children}</div>
  ),
  TileLayer: () => <div data-testid="mockTileLayer">Mock TileLayer</div>,
  useMap: jest.fn(() => ({
    setView: jest.fn(),
  })),
}));


describe('DataGrids/ForecastTileGroup/ForecastTile', () => {
  const renderComponent = (props: any, temperatureScale = TemperatureScale.Celsius) => {
    return render(
      <HistoricalContext.Provider value={{ temperatureScale } as any}>
        <ForecastTile {...props} />
      </HistoricalContext.Provider>
    );
  };

  it('matches ForecastTile component snapshot', () => {
    const props = { date: '2025-05-08', temperature: 22.5 };
    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });

  it('renders the correct day name and date', () => {
    const props = { date: '2025-05-08', temperature: 22.5 };
    renderComponent(props);
    expect(screen.getByTestId('dayName')).toHaveTextContent('Thursday');
    expect(screen.getByTestId('date')).toHaveTextContent('May 8');
  });

  it('renders the correct temperature in Celsius', () => {
    const props = { date: '2025-05-08', temperature: 22.5 };
    renderComponent(props, TemperatureScale.Celsius);
    expect(screen.getByTestId('temperature')).toHaveTextContent('22.5°C');
  });

  it('renders the correct temperature in Fahrenheit', () => {
    const props = { date: '2025-05-08', temperature: 22.5 };
    renderComponent(props, TemperatureScale.Fahrenheit);
    expect(screen.getByTestId('temperature')).toHaveTextContent('73°F'); // Assuming conversion is correct
  });
});