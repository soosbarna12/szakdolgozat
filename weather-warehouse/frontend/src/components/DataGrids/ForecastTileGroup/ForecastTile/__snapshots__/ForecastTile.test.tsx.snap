// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrids/ForecastTileGroup/ForecastTile matches ForecastTile component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1sj61kq-MuiPaper-root"
        data-testid="forecastTile"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-h9fkkg-MuiTypography-root"
          data-testid="dayName"
        >
          Thursday
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
          data-testid="date"
        >
          May 8
        </h6>
        <div
          class="MuiBox-root css-1uaifz8"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-ciltfl-MuiTypography-root"
            data-testid="temperature"
          >
            22.5
            <span
              style="font-size: 0.5em; vertical-align: super;"
            >
              °C
            </span>
          </h2>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            data-testid="precipitation"
          >
            Precipitation: 
            0.50
             mm/h
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            data-testid="pressure"
          >
            Pressure: 
            1013.00
             hPa
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1sj61kq-MuiPaper-root"
      data-testid="forecastTile"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-h9fkkg-MuiTypography-root"
        data-testid="dayName"
      >
        Thursday
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
        data-testid="date"
      >
        May 8
      </h6>
      <div
        class="MuiBox-root css-1uaifz8"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-ciltfl-MuiTypography-root"
          data-testid="temperature"
        >
          22.5
          <span
            style="font-size: 0.5em; vertical-align: super;"
          >
            °C
          </span>
        </h2>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          data-testid="precipitation"
        >
          Precipitation: 
          0.50
           mm/h
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          data-testid="pressure"
        >
          Pressure: 
          1013.00
           hPa
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
